generative/maisi/scripts/train_controlnet.py

```diff
@@ -0,0 +1,405 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+from pathlib import Path
+import time
+from datetime import timedelta
+import warnings
+
+warnings.simplefilter("ignore", UserWarning)
+
+import os
+import sys
+import copy
+import torch
+import torch.distributed as dist
+import torch.nn.functional as F
+from monai.config import print_config
+from monai.utils import first, set_determinism
+from monai.data import DataLoader, CacheDataset, partition_dataset
+from monai.networks.utils import copy_model_state
+from torch.cuda.amp import GradScaler, autocast
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.utils.tensorboard import SummaryWriter
+from monai.transforms import (
+    Compose,
+    EnsureTyped,
+    Lambdad,
+    LoadImaged,
+    Orientationd,
+)
+from monai.bundle import ConfigParser
+from utils import binarize_labels
+
+
+def setup_ddp(rank, world_size):
+    print(f"Running DDP diffusion example on rank {rank}/world_size {world_size}.")
```

Comment:
Since we are using the `logging` modules, shall we change the `print` to `logging.info`?

Also, the RankFilter in `monai` may be helpful to print message in selected rank (e.g. rank=0) in dist training case.

Reply:
thanks, refactored code to use logging and monai RankFilter.